process/block/preprocess/transactions.go

```diff
@@ -961,12 +962,12 @@ func (txs *transactions) createAndProcessMiniBlocksFromMe(
 		}
 
 		miniBlock.TxHashes = append(miniBlock.TxHashes, txHash)
-		numTxs := 1
-		if core.IsSmartContractAddress(tx.RcvAddr) {
+		txs.blockSizeComputation.AddNumTxs(1)
+		if receiverShardID != txs.shardCoordinator.SelfId() && core.IsSmartContractAddress(tx.RcvAddr) {
+			numCrossShardScCalls++
 			//we need to increment this as to account for the corresponding SCR hash
-			numTxs++
+			txs.blockSizeComputation.AddNumTxs(1)
```

Comment:
instead of adding one-by-one, we can add directly on line 973
```GO
txs.blockSizeComputation.AddNumTxs(numCrossShardScCalls)
```

Reply:
done